presto-main/src/main/java/io/prestosql/sql/analyzer/Analysis.java

```diff
@@ -316,6 +317,16 @@ public void setOrderByExpressions(Node node, List<Expression> items)
         return orderByExpressions.get(NodeRef.of(node));
     }
 
+    public void setLimit(Node node, Long rowCount)
+    {
+        limit.put(NodeRef.of(node), rowCount);
+    }
+
+    public Long getLimit(Node node)
```

Comment:
return primitive `long` and maybe check that the an entry exists in the `limit` map to avoid NPE in the off chance that there's a bug in the analyzer.

Reply:
I can't figure out how to store LIMIT ALL with primitive `long`.
With `Long` I used null value, but I suppose `Optional<Long>` would be the neatest solution with `Optional.empty` for LIMIT ALL.

Reply:
That seems reasonable, but I'd use `OptionalLong`, instead.

Reply:
@martint i was under impression that we prefer `Optional<Long>` over `OptionalLong` unless there is compelling (a.k.a. performance) reason to use the latter.
https://github.com/prestodb/presto/pull/11302#issuecomment-417803825 & https://github.com/prestodb/presto/pull/11302#issuecomment-417881712

Reply:
I guess it depends on how it’s being used. This is very localized. No strong preference either way in this case, so Optional is ok, too.